x-pack/plugins/event_log/server/event_log_client.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Observable } from 'rxjs';
+import { ClusterClient, SavedObjectsClientContract } from 'src/core/server';
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import { EsContext } from './es';
+import { IEventLogClient, IEvent } from './types';
+export type PluginClusterClient = Pick<ClusterClient, 'callAsInternalUser' | 'asScoped'>;
+export type AdminClusterClient$ = Observable<PluginClusterClient>;
+
+interface EventLogServiceCtorParams {
+  esContext: EsContext;
+  savedObjectsClient: SavedObjectsClientContract;
+}
+
+const optionalDateFieldSchema = schema.maybe(
+  schema.string({
+    validate(value) {
+      if (isNaN(Date.parse(value))) {
+        return 'Invalid Date';
+      }
+    },
+  })
+);
+
+export const findOptionsSchema = schema.object({
+  per_page: schema.number({ defaultValue: 10, min: 0 }),
+  page: schema.number({ defaultValue: 1, min: 1 }),
+  start: optionalDateFieldSchema,
+  end: optionalDateFieldSchema,
+});
+export type FindOptionsType = TypeOf<typeof findOptionsSchema>;
+
+// note that clusterClient may be null, indicating we can't write to ES
+export class EventLogClient implements IEventLogClient {
+  private esContext: EsContext;
+  private savedObjectsClient: SavedObjectsClientContract;
+
+  constructor({ esContext, savedObjectsClient }: EventLogServiceCtorParams) {
+    this.esContext = esContext;
+    this.savedObjectsClient = savedObjectsClient;
+  }
+
+  async findEventsBySavedObject(
+    type: string,
+    id: string,
+    options?: Partial<FindOptionsType>
+  ): Promise<IEvent[]> {
+    await this.savedObjectsClient.get(type, id);
```

Comment:
Might be worth a comment here to explain why it's important to do this even though we're not doing anything with the response. It would be nice to have api integration tests for this as well.

Reply:
Fair, though there's an explicit unit test that validates this and I consider that more reliable documentation than a comment, but happy to add one 🤷‍♂